Gulpfile.js

```diff
@@ -318,37 +318,57 @@ task("clean-tests").description = "Cleans the outputs for the test infrastructur
 
 const watchTests = () => watchProject("src/testRunner", cmdLineOptions);
 
-const buildRules = () => buildProject("scripts/tslint");
-task("build-rules", buildRules);
-task("build-rules").description = "Compiles tslint rules to js";
+const buildEslintRules = () => buildProject("scripts/eslint");
+task("build-eslint-rules", buildEslintRules);
+task("build-eslint-rules").description = "Compiles eslint rules to js";
 
-const cleanRules = () => cleanProject("scripts/tslint");
-cleanTasks.push(cleanRules);
-task("clean-rules", cleanRules);
-task("clean-rules").description = "Cleans the outputs for the lint rules";
+const cleanEslintRules = () => cleanProject("scripts/eslint");
+cleanTasks.push(cleanEslintRules);
+task("clean-eslint-rules", cleanEslintRules);
+task("clean-eslint-rules").description = "Cleans the outputs for the eslint rules";
+
+const runEslintRulesTests = () => runConsoleTests("scripts/eslint/built/tests", "mocha-fivemat-progress-reporter", /*runInParallel*/ false, /*watchMode*/ false);
+task("run-eslint-rules-tests", series(buildEslintRules, runEslintRulesTests));
+task("run-eslint-rules-tests").description = "Runs the eslint rule tests";
 
 const lintFoldStart = async () => { if (fold.isTravis()) console.log(fold.start("lint")); };
 const lintFoldEnd = async () => { if (fold.isTravis()) console.log(fold.end("lint")); };
-const lint = series([
-    lintFoldStart,
-    ...["scripts/tslint/tsconfig.json", "src/tsconfig-base.json"].map(project => {
-        const lintOne = () => {
-            const args = ["node_modules/tslint/bin/tslint", "--project", project, "--formatters-dir", "./built/local/tslint/formatters", "--format", "autolinkableStylish"];
-            if (cmdLineOptions.fix) args.push("--fix");
-            log(`Linting: node ${args.join(" ")}`);
-            return exec(process.execPath, args);
-        };
-        lintOne.dispayName = `lint(${project})`;
-        return lintOne;
-    }),
-    lintFoldEnd
-]);
+const eslint = (folder) => async () => {
+    const ESLINTRC_CI = ".eslintrc.ci.json";
+    const ESLINTRC = ".eslintrc.json";
+    const isCIEnv = process.env.CI === "true";

```

Comment:
May want to give a command line option to do this as well for convenience of running locally.

Reply:
@uniqueiniquity Added `--ci` option